analyses/npl/10_historical_event_timeline.md

```diff
@@ -0,0 +1,307 @@
+                    cname += f"_rp{rp}_p{percentile}"
+                if row[cname]:
+                    if not event_bools[event_type]:
+                        s = 'flood' if event_type == 'danger' else event_type + ' trigger'
+                        print(date, s)
+                        # Find lead time
+                        if event_type in ['readiness', 'action']:
+                            forecast_leadtimes = []
+                            for leadtime in LEADTIMES_BY_TRIGGER[event_type]:
+                                if row[f'forecast_lt{leadtime}_p{percentile}'] > rp_val:
+                                    forecast_leadtimes += [leadtime]
+                            df_events.loc[date, 'leadtimes'] = forecast_leadtimes
+                            #print(forecast_leadtimes)
+                    event_bools[event_type] = True
+                else:
+                    event_bools[event_type] = False
+```
+
+### Plots
+
+```python
+rp = 2
+x = ds_glofas_forecast_summary['Chisapani_v3'].sel(leadtime=4, percentile=[20, 25, 30, 35, 40, 45, 50])
+rp_val = df_return_period.loc[rp, 'Chisapani']
+
+
+fig, ax = plt.subplots()
+x.plot.line(hue='percentile', alpha=0.5)
+ax.set_xlim(np.datetime64('2009-08-10'), np.datetime64('2009-08-20'))
+ax.axhline(rp_val, c='k')
+ax.set_ylim(4000, 8000)
+```
+
+```python
+mpl.rcParams['hatch.linewidth'] = 0.5
+for station in STATIONS:
+    df_station = df_station_dict[station].copy().dropna(subset=["water_level"])
+    thresh = df_station_info.at[station, f'danger_level']
+    rp_val = df_return_period.loc[MAIN_RP, station]
+    years = df_station.index.year.unique()
+    fig, axs = plt.subplots(len(years), figsize=(8, 20))
+    for year, ax in zip(years, axs):
+        data = df_station[df_station.index.year == year]
+        # Plot everything
+        ax.plot(data.index, data.water_level, c='k', label='water level', alpha=0.75, lw=0.75)
+        ax2 = ax.twinx()
+        ax2.plot(data.index, data.river_discharge, c='C0', label='river_discharge', alpha=0.5, lw=0.5)
+        ax.plot([], [], 'C0-', label='river discharge', alpha=0.5, lw=0.5)
+        ax.axhline(y=thresh, c='C3', label='danger / 1 in 2 y', alpha=0.75, lw=0.5)
+        
+        y0 = -0.05 * thresh
+        y1 = thresh*1.2
+        ax.set_ylim(y0, y1)
+        ax2.set_ylim(-0.05 * rp_val, rp_val*1.2)
+        ax.set_xticklabels([])
+        ax.text(data.index[0], thresh + 0.2, f"{year}")
+        ax.grid()
+        # Add the activations
+        for event_type, hatch, colour, lw in zip(['readiness', 'action','danger'], 
+                                             ['/////', '\\\\\\\\\\', '.....'], 
+                                             ['C2', 'C1', 'C6'],
+                                             [0.5, 0.5, 2]):
+            cname = f'event_{event_type}'
+            if event_type != "danger":
+                cname += f"_rp{MAIN_RP}"
+            ax.fill_between(data.index, y0, y1, where=data[cname], 
```

Comment:
Am just getting ```KeyError: 'event_readiness_rp2'``` when trying to rerun this notebook.

Reply:
Ah sorry, this has been fixed in #131 which was supposed to be the main branch for this file. Not sure why it's showing up here!